pallets/parachain-staking/src/lib.rs

```diff
@@ -2087,11 +2101,15 @@ pub mod pallet {
 		/// delegator and tries to compound a specified percent of it back towards the delegation.
 		/// If a scheduled delegation revoke exists, then the amount is only minted, and nothing is
 		/// compounded. Emits the [Compounded] event.
+		/// already_called_for_delegator: bool is used to indicate if this function has already been
+		/// called for a given delegator by the callee. This is used to avoid over-estimating
+		/// weight, since subsequent calls to this function will not increase the pov size.
 		pub fn mint_and_compound(
 			amt: BalanceOf<T>,
 			compound_percent: Percent,
 			candidate: T::AccountId,
 			delegator: T::AccountId,
+			already_called_for_delegator: bool,
```

Comment:
```suggestion
			already_called_for_candidate: bool,
```
The discrepancy arises due to being called for the same candidate